test/DynamoCoreWpfTests/GraphLayoutTests.cs

```diff
@@ -36,6 +37,25 @@ public void GraphLayoutEmpty()
         }
 
         [Test]
+        public void GraphLayoutMarkFileAsDirty()
+        {
+            // A graph with one node
+            OpenModel(GetDynPath("GraphLayoutOneNode.dyn"));
+            IEnumerable<NodeModel> nodes = ViewModel.CurrentSpace.Nodes;
+
+            Assert.IsFalse(ViewModel.CurrentSpace.HasUnsavedChanges);
+            ViewModel.DoGraphAutoLayout(null);
+            Assert.IsFalse(ViewModel.CurrentSpace.HasUnsavedChanges);
```

Comment:
Yes, otherwise users might lose their changes when closing a file after running graph layout. `HasUnsavedChanges` is a workspace property and will not re-execute the graph.

Ref: [MAGN-8302](http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-8302)